SConstruct

```diff
@@ -215,52 +266,138 @@ Default(environ.get('M5_DEFAULT_BINARY', 'build/ARM/gem5.debug'))
 #
 ########################################################################
 
-# helper function: find last occurrence of element in list
-def rfind(l, elt, offs = -1):
-    for i in range(len(l)+offs, 0, -1):
-        if l[i] == elt:
-            return i
-    raise ValueError("element not found")
+kconfig_actions = (
+    'defconfig',
+    'guiconfig',
+    'listnewconfig',
+    'menuconfig',
+    'oldconfig',
+    'olddefconfig',
+    'savedefconfig',
+    'setconfig',
+)
+
+Help("""
+Kconfig:
+        In addition to the default configs, you can also create your own
+        configs, or edit one that already exists. To use one of the kconfig
+        tools with a particular directory, use a target which is the directory
+        to configure, and then the name of the tool. For example, to run
+        menuconfig on directory build/foo/bar, run:
+
+        scons menuconfig build/foo/bar
```

Comment:
> `build/foo/bar`

All of these examples make it seem like you have to use `build` as the root of the directory structure. I don't think this is the case, correct?

Reply:
After the kconfig changed, the user can build gem5 without need to follow the naming structure of `build/<build_opts>/gem5.opt`. Instead, they can use the customized the build namiing like `gem5_build/gem5.opt`, `foo_bar/gem5.debug` etc.